File: projects/practica_docker/volumes/quasar/src/model/apps/convalidacions/ItemConvalidacio.ts
import {CategoriaConvalidacio} from "src/model/apps/convalidacions/CategoriaConvalidacio";

export interface ItemConvalidacio {
  id?: number;
  codi: string;
  nom: string;
  categoria: CategoriaConvalidacio;
  composa?: Array<ItemConvalidacio>;
  label?: string;
  value?: string;
}

File: projects/practica_docker/volumes/java/src/main/java/cat/iesmanacor/gestibgsuite/model/apps/convalidacions/ItemConvalidacio.java
package cat.iesmanacor.gestibgsuite.model.apps.convalidacions;

import com.fasterxml.jackson.annotation.JsonBackReference;
import lombok.Data;
import lombok.EqualsAndHashCode;

import javax.persistence.*;
import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name = "im_app_convalidacions_item")
public @Data class ItemConvalidacio {
    @Id
    @Column(name = "iditem")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long iditem;

    @Column(name = "codi", nullable = true, length = 2048)
    private String codi;

    @Column(name = "nom", nullable = false, length = 2048)
    private String nom;

    @ManyToOne(optional = false)
    private CategoriaConvalidacio categoria;

    @OneToMany
    private Set<ItemConvalidacio> composa = new HashSet<>();
}
